src/segment_memory.go

```diff
@@ -31,6 +40,15 @@ func (n *memory) string() string {
 func (n *memory) init(props *properties, env environmentInfo) {
 	n.props = props
 	n.env = env
-	n.TotalMemory = mem.TotalMemory()
-	n.FreeMemory = mem.FreeMemory()
+	if props.getString(MemoryType, "physical") == "physical" {
```

Comment:
Can we swap the `if else` for an early return? See [object calisthenics](https://williamdurand.fr/2013/06/03/object-calisthenics/#2-dont-use-the-else-keyword).